lib/index.coffee

```diff
@@ -15,35 +15,38 @@ create_redis_client = _.memoize (host, port, password, log) ->
 
   # Client emits an error every time it tries to reconnect and fails. Only emit an error once.
   _redis.once 'error', (err) ->
-    log "RESERVE: Error connecting to REDIS: #{host}:#{port}.", err
+    logger.errorD "redis-error", { host, port, err }
     throw err
   _redis
 
 , (host, port, password) -> "#{host}:#{port}:#{password}"
 
 module.exports = class ReserveResource
-  constructor: (@by, @host, @port, @heartbeat_interval, @lock_ttl, @log=console.error, @password='') ->
+  constructor: (@by, @host, @port, @heartbeat_interval, @lock_ttl, @password='') ->
     @_lost_reservation = false
+    @identifer = process.env.JOB_ID or process.pid
+    @logger = new kayvee.logger(@by)
+    @logger.globals.job_id = process.env.JOB_ID
```

Comment:
As much as I want to be able to search redis-reservation logs by job_id, this seems too specific to our internal usage, considering this repo is open-source (although I don't know how many external users it actually has).

Could you allow passing in an optional logger to the constructor, so you can init the logger with specific configuration outside this library?